src/spikeinterface/core/sorting_tools.py

```diff
@@ -440,3 +441,176 @@ def generate_unit_ids_for_merge_group(old_unit_ids, merge_unit_groups, new_unit_
             raise ValueError("wrong new_id_strategy")
 
     return new_unit_ids
+
+
+### SPLITTING ZONE ###
+def apply_splits_to_sorting(sorting, unit_splits, new_unit_ids=None, return_extra=False, new_id_strategy="append"):
+    spikes = sorting.to_spike_vector().copy()
+
+    # here we assume that unit_splits split_indices are already full.
+    # this is true when running via apply_curation
```

Comment:
I would make a check no ? Or having an option for this to skip the check when coming from analyzer split

Reply:
added